doc/python/colorscales.md

```diff
@@ -84,11 +121,154 @@ import plotly.express as px
+                                                     (0.33, "green"), (0.66, "green"),
+                                                     (0.66, "blue"),  (1.00, "blue")])
+fig.show()
+```
+
+### Explicitly setting a Color Range
+
+When using the range of the input data as the color range is inappropriate, for example when producing many figures which must have comparable color ranges, or to clip the color range to account for outliers, the Plotly Express `range_color` argument can be used. Here we clip the top of the color range above the lower range of the data and extend it below the lower range of the data:
+
+```python
+import plotly.express as px
+df = px.data.iris()
+fig = px.scatter(df, x="sepal_width", y="sepal_length", color="sepal_length", range_color=[5,8])
+
+fig.show()
+```
+
+### Setting the Midpoint of a Color Range for a Diverging Colorscale
+
+Diverging colorscales have a well-defined midpoint color, and are best-used when that midpoint is mapped to a meaningful data value. The `color_continuous_midpoint` argument to most Plotly Express functions is used for this. It cannot be used with `range_color` because setting it forces the color range to be centered on the midpoint while including the entire dataset. This means that for asymmetric data distributions, not all colors in the color scale will appear in the figure.
+
+For example, a diverging colorscale could be used to highlight points with a higher and lower value than the median in a choropleth map like this:
+
+```python
+import plotly.express as px
+
+df = px.data.gapminder().query("year == 2007")
+avg_lifeExp = (df['lifeExp']*df['pop']).sum()/df['pop'].sum()
+
+fig = px.choropleth(df, locations="iso_alpha", color="lifeExp",
+                    color_continuous_scale=px.colors.diverging.BrBG,
+                    color_continuous_midpoint=avg_lifeExp,
+                    title="World Average Life Expectancy in 2007 in years was %.1f" % avg_lifeExp)
+fig.show()
+```
+
+### Hiding or Customizing the Plotly Express Color Bar
+
+Plotly Express binds all traces to [`layout.coloraxis`](/python/reference/#layout-coloraxis), rather than using trace-specific color axes. This means that the color bar can configured there, for example it can be hidden:
+
+```python
+import plotly.express as px
+df = px.data.tips()
+fig = px.density_heatmap(df, x="total_bill", y="tip", title="No color bar on this density plot")
+
+fig.update_layout(coloraxis_showscale=False)
+
+fig.show()
+```
+
+You can also configure the title, size, placement and tick marks and labels on a color bar:
+
+```python
+import plotly.express as px
+df = px.data.tips()
+fig = px.density_heatmap(df, x="total_bill", y="tip", title="Customized color bar on this density plot")
+
+fig.update_layout(coloraxis_colorbar=dict(
+    title="Number of Bills per Cell",
+    thicknessmode="pixels", thickness=50,
+    lenmode="pixels", len=200,
+    yanchor="top", y=1,
+    ticks="outside", ticksuffix=" bills",
+    dtick=5
```

Comment:
Suggestion: add
`tickvals=[0, 10, 20, 30]`
here, this is something that users often want to configure.
Maybe also add a link to the axes tutorial which has a number of examples about ticks.

Reply:
oops I had not seen that it was done in the next example, please ignore this suggestion